mesonbuild/modules/pkgconfig.py

```diff
@@ -136,6 +138,34 @@ def generate(self, state, args, kwargs):
         pub_reqs = mesonlib.stringlistify(kwargs.get('requires', []))
         priv_reqs = mesonlib.stringlistify(kwargs.get('requires_private', []))
         conflicts = mesonlib.stringlistify(kwargs.get('conflicts', []))
+
+        def parse_variable_list(stringlist):
+            reserved = ['prefix', 'libdir', 'includedir']
+            variables = []
+            for var in stringlist:
+                l = var.split('=')
+                if len(l) < 2:
+                    raise mesonlib.MesonException('Variables must be in \'name=value\' format')
+
+                # foo=bar=baz is ('foo', 'bar=baz')
+                k, v = l[0], '='.join(l[1:])
```

Comment:
It is simpler to do `l = var.split('=', 1)` instead. Then you can do `k = l[0]` and `v = l[1]`.